### PR TITLE
Handling Uphold's CDD (customer due diligence) process.

### PR DESCRIPTION
**Pull request by szilardszaloki**
Resolves https://github.com/brave/brave-browser/issues/20427, resolves https://github.com/brave/brave-browser/issues/20497.

- introducing a modal (`upholdCustomerDueDiligenceRequiredModal`) that tells the user to complete the CDD survey on Uphold
- parsing out `verifications.customerDueDiligence.status` from the response of `GET https://api.uphold.com/v0/me`
- sending the user to `NOT_CONNECTED`/`DISCONNECTED_VERIFIED` if CDD is required for them
- showing `upholdCustomerDueDiligenceRequiredModal` (and the `kWalletDisconnected` notification, if they're already `VERIFIED`)
- adding test cases

`upholdCustomerDueDiligenceRequiredModal`:
upholdCustomerDueDiligenceRequiredModal
`Learn More` links to `https://wallet.uphold.com/customer-due-diligence`.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
These tests need an Uphold account that requires CDD.
### Testing linkage:
- `PENDING` `==>` `NOT_CONNECTED` (during linking attempt): only modal – no notification (since they haven't been verified)

https://user-images.githubusercontent.com/5390451/149630180-11d433b5-6b3c-4f09-b5fe-a8a2356f7125.mp4

### Upgrade tests:
Make sure you're on `brave://rewards` when upgrading to the new version!
- `VERIFIED` `==>` `DISCONNECTED_VERIFIED` (on `Rewards` page): modal + notification

https://user-images.githubusercontent.com/5390451/149630233-e6d4bdda-da02-4d7a-856c-1fe8ebee4a1c.mp4

Make sure you're _not_ on `brave://rewards` when upgrading to the new version!
- `VERIFIED` `==>` `DISCONNECTED_VERIFIED` (on `Rewards` panel): only notification – no modal (since they're not on `Rewards` page)

https://user-images.githubusercontent.com/5390451/149630243-1b79fb4c-e816-4eb9-84f8-c5dddaff1565.mp4

As far as the spinner issue, see https://github.com/brave/brave-browser/issues/20529.
Android doesn't have the `wallet_disconnected` notification (`ExternalWalletDisconnectedNotification`) implemented, see https://github.com/brave/brave-browser/issues/20603.

### CODE REVIEW

File: components/brave_rewards/resources/page/components/settingsPage.tsx [hunk 1]
```diff
@@ -291,6 +291,18 @@ class SettingsPage extends React.Component<Props, State> {
...
+          <ModalRedirect
+            id={'redirect-modal-uphold-customer-due-diligence-required'}
+            errorText={getLocale('redirectModalUpholdCustomerDueDiligenceRequiredText')}
+            titleText={getLocale('redirectModalUpholdCustomerDueDiligenceRequiredTitle')}
+            learnMore={'https://wallet.uphold.com/customer-due-diligence'}
+            buttonText={getLocale('redirectModalClose')}
+            walletType={walletType}
+            onClick={this.actions.hideRedirectModal}
+          />
+        )
```
Comment (emerick) [inline components/brave_rewards/resources/page/components/settingsPage.tsx:305]: I think we mentioned this in the meeting today, but any changes made to `components/brave_rewards/resources/page` should be made to `components/brave_rewards/resources/android_page` as well.

File: components/brave_rewards/resources/page/reducers/wallet_reducer.ts [hunk 2]
```diff
@@ -203,6 +203,11 @@ const walletReducer: Reducer<Rewards.State | undefined> = (state: Rewards.State,
         break
       }
 
+      if (action.payload.result === 43) { // type::Result::UPHOLD_CUSTOMER_DUE_DILIGENCE_REQUIRED
+        state.ui.modalRedirect = 'upholdCustomerDueDiligenceRequiredModal'
+        break
+      }
+
```
Comment (emerick) [inline components/brave_rewards/resources/page/reducers/wallet_reducer.ts:210]: Same comment as above.

**Review by emerick — APPROVED** [review]
LGTM, we should just make sure to create an issue for the Android piece of this.

File: components/brave_rewards/resources/page/components/settingsPage.tsx (hunk 1 above)
Comment (szilardszaloki) [reply components/brave_rewards/resources/page/components/settingsPage.tsx:305]: Yep, just pushed them.
@emerick Would you mind helping me with testing this on Android? I don't have an Android device at hand.

Comment (emerick) [reply components/brave_rewards/resources/page/components/settingsPage.tsx:305]: Sure, ping me whenever you're ready!

**Comment by szilardszaloki** [comment]
`SonarCloud` is reporting duplicated lines due to how we currently structure code in `settingsPage.tsx` and `wallet_reducer.ts`.

**Comment by LaurenWags** [comment]
Verification in progress with

```
Brave | 1.36.63 Chromium: 97.0.4692.99 (Official Build) nightly (x86_64)
-- | --
Revision | d740da257583289dbebd2eb37e8668928fac5ead-refs/branch-heads/4692@{#1461}
OS | macOS Version 11.6.1 (Build 20G224)
```

<details>
<summary>Scenario 1: PENDING ==> NOT_CONNECTED - InProgress</summary>

</details>

**Pull request merged**